js/core.js

```diff
@@ -666,7 +666,46 @@ var ERMrest = (function (module) {
             }
 
             return result;
-        }
+        },
+
+        // figure out if Table is pure and binary association table.
+        // binary: Has 2 outbound foreign keys and there is a composite key constraint combining all the columns from both foreign keys.
+        // pure: There are no other columns apart from the foreign key columns and columns that are part of key and have serial4 type.
+        _isPureBinaryAssociation: function () {
+            if(this._isPureBinaryAssociation_cached === undefined) {
+                this._isPureBinaryAssociation_cached = this._computePureBinaryAssociation();
+            }
+            return this._isPureBinaryAssociation_cached;
+        },
+
+        _computePureBinaryAssociation: function () {
+            if (this.referredBy.length() > 0 || this.foreignKeys.length() != 2) {
+                return false; // not binary
+            }
+
+            var fkColset = new ColSet(this.foreignKeys.colsets().reduce(function(res, colset){
+                return res.concat(colset.columns);
+            }, [])); // set of foreignkey columns
+
+            var tempKeys = this.keys.all().filter(function(key) {
+                var keyCols = key.colset.columns;
+                return !(keyCols.length == 1 && keyCols[0].type.name == "serial4"  && !(keyCols[0] in fkColset.columns));
+            }); // the key that should contain foreign key columns.
+
+            if (tempKeys.length != 1 || !fkColset._equals(tempKeys[0].colset)) {
+                return false; // not binary
+            }
+
+            // check for purity:
+            for (var i = 0, col; i < this.columns.length(); i++) {
```

Comment:
The temp keys started as all keys and removed the trivial serial keys.

Then, you check that there is only one temp key left and that it is
exactly the foreign key columns (that's what the colset equality test
means).

There is nothing else that needs to be checked since this covered
everything needed for "pure" binary association.  The impure parts,
aside from trivial serial keys we removed earlier, would be left as
extra temp keys and/or a final temp key that fails the colset equality
test.